cmd/observer/node_builder/observer_builder.go

```diff
@@ -994,58 +990,27 @@ func (builder *ObserverServiceBuilder) enqueueRPCServer() {
 			return nil, err
 		}
 
-		client, err := builder.initUpstreamClient()
+		// upstream access node forwarder
+		forwarder, err := apiproxy.NewFlowAccessAPIForwarder(builder.upstreamIdentities, builder.apiTimeout)
 		if err != nil {
 			return nil, err
 		}
 
+		proxy := &apiproxy.FlowAccessAPIRouter{
+			Logger:          builder.Logger,
+			Metrics:         metrics.NewObserverCollector(),
+			Upstream:        forwarder,
+			AccessAPIServer: engineBuilder.Handler(),
```

Comment:
I'd like to directly create the local handler here instead of calling `engineBuilder.Handler()` to access the local handler created [here](https://github.com/onflow/flow-go/pull/2920/files#diff-082fe5eb166eeea14093fceec4c27c8ac626018c23876d187b99f32eef131c21R28). Then overwriting that with `WithNewHandler(proxy)`. It's just really hacky.